pkgs/by-name/si/siyuan/package.nix

```diff
@@ -20,8 +24,17 @@ let
   };
 
   platformId = platformIds.${stdenv.system} or (throw "Unsupported platform: ${stdenv.system}");
+
+  desktopEntry = makeDesktopItem {
+    name = "siyuan";
+    desktopName = "Siyuan";
+    comment = "Personal knowledge management system";
+    icon = "siyuan";
+    exec = "siyuan %u";
+    categories = [ "Office" ];
+  };
 in
-stdenv.mkDerivation (finalAttrs: {
+stdenv.mkDerivation (finalAttrs: rec {
```

Comment:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

Reply:
Note:
`finalAttrs` is used for recursion.
You can achieve recursion by just writing `finalAttrs.foobar` instead of just writing `foobar`.